### DISCUSSION TITLE
TypeError: RepositoryClass is not a constructor

### DISCUSSION THREAD

**Discussion by wujingquan** [mikro-orm/mikro-orm#5223] [General]
### Describe the bug

![image](https://github.com/mikro-orm/mikro-orm/assets/22289015/2b4eedb1-eccd-485d-9af9-5abf50e8b67f)

file: ExtendedEntityRepository.ts
```
import { EntityManager, EntityRepository, AnyEntity } from '@mikro-orm/mysql';

export class ExtendedEntityRepository<
  T extends object,
> extends EntityRepository<T> {
  persist(entity: AnyEntity | AnyEntity[]): EntityManager {
    return this.em.persist(entity);
  }

  async persistAndFlush(entity: AnyEntity | AnyEntity[]): Promise<void> {
    await this.em.persistAndFlush(entity);
  }

  remove(entity: AnyEntity): EntityManager {
    return this.em.remove(entity);
  }

  async removeAndFlush(entity: AnyEntity): Promise<void> {
    await this.em.removeAndFlush(entity);
  }

  async flush(): Promise<void> {
    return this.em.flush();
  }
}

```

![image](https://github.com/mikro-orm/mikro-orm/assets/22289015/c1c244c6-a3d7-476d-b74d-0dbdf8d72637)
```
import { MikroOrmModule } from '@mikro-orm/nestjs';
import { ConfigModule, ConfigService } from '@nestjs/config';
import { SqlHighlighter } from '@mikro-orm/sql-highlighter';
import { Module } from '@nestjs/common';
import { AppController } from './app.controller';
import { AppService } from './app.service';
import { Example } from './entities/Example';
import { RoleModule } from './role/role.module';
import { ApiModule } from './api/api.module';
import { UserModule } from './user/user.module';
import { MySqlDriver, Options } from '@mikro-orm/mysql';
import { ExtendedEntityRepository } from './common/ExtendedEntityRepository';

@Module({
  imports: [
    MikroOrmModule.forRootAsync({
      imports: [ConfigModule.forRoot()],
      inject: [ConfigService],
      useFactory: (config: ConfigService): Options => {
        return {
          strict: true,
          driver: MySqlDriver,
          host: config.get('DB_HOST'),
          dbName: config.get('DB_NAME'),
          user: config.get('DB_USERNAME'),
          password: config.get('DB_PASSWORD'),
          entities: ['./dist/entities'],
          entitiesTs: ['./src/entities'],
          debug: Boolean(config.get('DB_DEBUG')),
          highlighter: new SqlHighlighter(),
          entityRepository: () => ExtendedEntityRepository,
        };
      },
    }),
    MikroOrmModule.forFeature([Example]),
    RoleModule,
    ApiModule,
    UserModule,
  ],
  controllers: [AppController],
  providers: [AppService],
})
export class AppModule {}

```

### Reproduction

1. Setup NestJS application
2. Follow the steps below to operate
![image](https://github.com/mikro-orm/mikro-orm/assets/22289015/6d6a3aa2-b2a3-4fdb-b39f-9962811fd9d7)

### What driver are you using?

@mikro-orm/mysql

### MikroORM version

6.1.0

### Node.js version

20.11.0

### Operating system

windows 10

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
It's not supposed to be a callback, fixed the docs [here](https://github.com/mikro-orm/mikro-orm/commit/3550d853a8b50892c417c6c66ac1030ae31f441d).

**Reply by thatsgolden** [reply]
I am using v7 version, got the same error, the [v7 doc](https://mikro-orm.io/docs/repositories#generic-custom-repository) says to use a callback, is that wrong?

**Reply by B4nan** [reply]
It's still not a callback: https://mikro-orm.io/api/core/interface/Options#entityRepository

**Reply by thatsgolden** [reply]
thanks, It is a good thing to confirm that the document is incorrect, because I tried using Class directly and there was no problem.

**Reply by B4nan** [reply]
Fixed via #7428
